### PR TITLE
feat: Add AOST queries

### LINKED ISSUE #281: support for AOST queries
CockroachDB has support for AS OF SYSTEM TIME timestamp queries. This can be prefixed to any SELECT query to query using the systemtime and get a historical view of the data. More info here:
https://www.cockroachlabs.com/docs/v1.1/as-of-system-time

It would be good to have support for AOST queries in ActiveRecord. Some potential approaches using select examples from [here](https://api.rubyonrails.org/v7.0.6/classes/ActiveRecord/QueryMethods.html#method-i-select):

```ruby
Model.select(:field).aost(some_timestamp)
Client.where("orders_count = '2'").aost(some_timestamp)
```

I am not 100% certain on if adding .aost method will break things or would be an antipattern in ActiveRecord/Ruby. cc: @BuonOmo for guidance on that.

### PR DESCRIPTION
**Pull request by BuonOmo**
Add _As Of System Time_ support for models.

See https://www.cockroachlabs.com/docs/stable/as-of-system-time

Informs #281

### CODE REVIEW

File: test/cases/arel/nodes_test.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
```
Comment (BuonOmo) [inline test/cases/arel/nodes_test.rb:1]: This can be removed on next rails update as it was fixed upstream

Comment (rafiss) [reply test/cases/arel/nodes_test.rb:1]: could a comment be added that mentions this?

**Review by rafiss — APPROVED** [review]
this looks good to me!

CRDB also supports using it like ` ... AS OF SYSTEM TIME '-10s'` (allowing any negative interval), as well as a few special functions for the parameter, described here: https://www.cockroachlabs.com/docs/stable/as-of-system-time#parameters

i think we can merge this PR, but could those other variants be supported too?

File: test/cases/arel/nodes_test.rb (hunk 1 above)
Comment (BuonOmo) [reply test/cases/arel/nodes_test.rb:1]: There is also one (see the excluded folder, it is mentionning the given PR). But I can do more for sure :)

**Comment by BuonOmo** [comment]
@rafiss I saw those, and since activesupport is also providing such features (`10.seconds.ago` is valid ruby !) I was thinking of keeping it simple. Also for the reason that the feature for now only work out of transaction (see my comment in the issue #281) I don't think it is worth making it more complex for now. What do you think ?

As for the other functions, I completely overlooked them, lets keep the AOST issue open and discuss the extended scope over there if that's ok with you!

**Comment by rafiss** [comment]
sounds great, thanks!

**Comment by grantpaulson6** [comment]
When do you guys think this will be released?

Also, just wanted to bump the comment about special functions for the AOST query, would be a nice to have

**Comment by BuonOmo** [comment]
@rafiss I don't see any particular reason not to release. WDYT?

**Pull request merged**